src/components/ContractTab/function-interface-utils.js

```diff
@@ -1,81 +1,436 @@
+ * @param {string} type
+ * @returns {boolean}
+ */
 function parameterTypeIsBooleanArray(type) {
-    return /^bool\[\d*]/.test(type);
+    return /^bool\[\d*]$/.test(type);
+}
+
+/**
+ * Given a function interface type, returns true iff that type is bytes, e.g. bytes32 or bytes[]
+ * @param {string} type
+ * @returns {boolean}
+ */
+function parameterTypeIsBytes(type) {
+    return /^bytes/.test(type)
+}
+
+/**
+ * Given a function interface type, returns true iff that type is signed integer, e.g. int32
+ * @param {string} type
+ * @returns {boolean}
+ */
+function parameterTypeIsSignedInt(type) {
+    return /^int\d+$/.test(type)
 }
 
+/**
+ * Given a function interface type, returns true iff that type array of signed integers, e.g. int32[]
+ * @param {string} type
+ * @returns {boolean}
+ */
+function parameterTypeIsSignedIntArray(type) {
+    return /^int\d+\[\d*]/.test(type)
+}
+
+/**
+ * Given a function interface type, returns true iff that type is string
+ * @param {string} type
+ * @returns {boolean}
+ */
 function parameterTypeIsString(type) {
     return type === 'string';
 }
 
+/**
+ * Given a function interface type, returns true iff that type is array of string
+ * @param {string} type
+ * @returns {boolean}
+ */
+function parameterTypeIsStringArray(type) {
+    return /^string\[\d*]$/.test(type);
+}
+
+/**
+ * Given a function interface type, returns true iff that type represents unsigned integer, e.g. uint32
+ * @param {string} type
+ * @returns {boolean}
+ */
+function parameterTypeIsUnsignedInt(type) {
+    return /^uint\d+$/.test(type)
+}
+
+/**
+ * Given a function interface type, returns true iff that type represents unsigned integer array, e.g. uint32[]
+ * @param {string} type
+ * @returns {boolean}
+ */
+function parameterTypeIsUnsignedIntArray(type) {
+    return /^uint\d+\[\d*]$/.test(type)
+}
+
+
+
+/**
+ * Given a function interface type, returns true iff that type represents an array of any kind, e.g. string[] or uint8[]
+ * @param {string} type
+ * @returns {boolean}
+ */
+function parameterIsArrayType(type) {
+    return /\[\d*]$/.test(type) || parameterTypeIsBytes(type);
+}
+
+/**
+ * Given a function interface type, returns true iff that type represents an integer, either signed or unsigned
+ * @param {string} type
+ * @returns {boolean}
+ */
+function parameterIsIntegerType(type) {
+    return /int\d+$/.test(type);
+}
+
+/**
+ * Validate integer-size props for input components, e.g. 256 for uint256
+ *
+ * @param {unknown} prop - the prop to be validated
+ * @param {boolean} signed - whether the target integer is signed
+ *
+ * @return {boolean}
+ */
+function integerSizeValidator(prop, signed) {
+    if (![true, false].includes(signed)) {
+        throw 'Invalid parameter - "signed" must be boolean';
+    }
+
+    const propIsNumber = Number.isInteger(+prop);
+    const propIs8Multiple = +prop % 8 === 0;
+
+    const max = signed ? 128 : 256;
+    const propIsInRange = +prop >= 8 && +prop <= max;
```

Comment:
I didn't know bout the existence of this `+` unary operator. I would have solved this problem by creating a new auxiliary const with the numeric value extracted using some deductive logic (like asking `typeof` and using `parseInt`).